src/tests/ResourceManagerTest.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <Corrade/Utility/Directory.h>
+#include <Magnum/EigenIntegration/Integration.h>
+#include <gtest/gtest.h>
+#include <string>
+
+#include "esp/assets/ResourceManager.h"
+#include "esp/gfx/Renderer.h"
+#include "esp/gfx/WindowlessContext.h"
+#include "esp/scene/SceneManager.h"
+
+#include "configure.h"
+
+namespace Cr = Corrade;
+
+using esp::assets::ResourceManager;
+using esp::scene::SceneManager;
+
+const std::string dataDir = Cr::Utility::Directory::join(SCENE_DATASETS, "../");
+
+TEST(ResourceManagerTest, createJoinedCollisionMesh) {
+  esp::gfx::WindowlessContext::uptr context_ =
+      esp::gfx::WindowlessContext::create_unique(0);
+
+  std::shared_ptr<esp::gfx::Renderer> renderer_ = esp::gfx::Renderer::create();
```

Comment:
Are we experiencing any segfault if we do not initialize the renderer_ here? (I did not see it used anywhere in the code.)